solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -751,16 +729,8 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
             return;
         }
 
-        // remove base token liquidity
-
         // calculate the amount of pool tokens required for liquidation
-        // note that the amount is doubled since it's not possible to liquidate one reserve only
-        Fraction memory poolRate = poolTokenRate(liquidity.poolToken, liquidity.reserveToken);
-        uint256 poolAmount = targetAmount.mul(poolRate.d).mul(2).div(poolRate.n);
-
-        // limit the amount of pool tokens by the amount the system holds
-        uint256 systemBalance = store.systemBalance(liquidity.poolToken);
-        poolAmount = poolAmount > systemBalance ? systemBalance : poolAmount;
+        uint256 poolAmount = liquidationAmount(liquidity.poolToken, liquidity.reserveToken, 0, targetAmount, false);
```

Comment:
notice that in the previous version of this function, we didn't add the pool amount to the system balance, while `liquidationAmount` does it, which will produce different results.

Reply:
We actually did. Please check the previous version, you'll se it there.

Reply:
see my comment above re. the functional error.

Reply:
See mine too.